rfcs/text/responsiveGrid.md

```diff
@@ -0,0 +1,152 @@
+- Start Date: 2019-12-11
+
+# Table of contents
+
+- [Summary](#summary)
+- [Basic example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+  - [CSS Grid Layout](#css-grid-layout)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [CSS Flexbox](#css-flexbox)
+  - [CSS Framework (Bootstrap or Foundation)](#css-framework-bootstrap-or-foundation)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+
+# Summary
+
+A responsive grid system is a layout component that provides guidance on how components should be positioned within a user interface, adapting to screen size and orientation. We, the Carbon team, would like to implement the grid system designed by Sage UI designers.
+
+# Basic example
+
+http://designsystem.sage.com/product/invoice/
+
+# Motivation
+
+The grid layout is intended to solve a number of problems that are often experienced when developing complex web applications:
+1. CSS floats are not well suited to application layout.
+2. Layouts that use a combination of tables, JavaScript and floated elements are often brittle and can behave unexpectedly.
+3. Fixed layouts that behave predictably cannot take advantage of the entire screen real estate.
+
+A well designed grid layout will be:
+1. Consistent, which helps users navigate and understand where information will be found.
+2. Responsive, which gives users a predictable experience across multiple devices with different screen sizes.
+3. Adaptable, enabling new design themes to be introduced more quickly as designs evolve over time.
+
+# Detailed design
+
+By dividing the available space into predictably-sized modules (both vertically and horizontally), complex layouts that don't have fixed dimensions can be controlled and components can be positioned allowing a clean separation of content and style. Once the underlying structure of a design is mapped, breakpoints can be used to control how each component responds at given widths.
+
+For maximum flexibility, the Design System responsive grid uses 12 equal-width columns. 12 is divisible into 12, 6, 4, 3, 2 or 1 equal-width columns, giving maximum possible flexibility for a layout.
+
+For full details of the proposal, read the [Sage Design System Grid](http://designsystem.sage.com/foundations/grid/).
+
+
+
+To avoid introducing a breaking change, and to give developers the freedom to choose whether or not to use the component, a new `Grid` Component will be developed, that can be used at the same level as the `<AppWrapper />`, as a grid wrapper for components. 
+The `<GridLayout />` will supply props to define breakpoints and number of columns, and will be styled using CSS Grid Layout.
+
+Code example:
+```jsx
+import AppWrapper from 'carbon-react/lib/components/app-wrapper';
```

Comment:
indentation is off here (either this should be indented by one or everything below it should be aligned to this)